src/main/java/edu/hawaii/its/api/service/UpdateTimestampService.java

```diff
@@ -18,21 +21,23 @@
  */
 @Service("timestampService")
 public class UpdateTimestampService {
-
     public static final Log logger = LogFactory.getLog(MembershipService.class);
 
-    public GroupingTimestampResult update(GroupingResult groupingResult) {
+    @Autowired
+    public GroupPathService groupPathService;
+
+    public GroupingTimestampResults update(GroupingResult groupingResult) {
         if (groupingResult.getResultCode().equals("SUCCESS")) {
-            return updateLastModifiedTimestamp(groupingResult.getGroupPath());
+            List<String> groupPaths = groupPathService.getGroupingGroupPaths(groupingResult.getGroupPath());
+            return updateLastModifiedTimestamp(groupPaths);
         }
-        return new GroupingTimestampResult();
+        return new GroupingTimestampResults();
```

Comment:
Sorry for the back and forth changes, it would be better to move the logic out of GroupPathService.getGroupingGroupPaths and back into this update function because updating only the owners group path is specific to timestamps. Also make sure there is test coverage for the logic. Thanks for your hard work on this ticket, this should be the last change to make.